app/views/homepage/_location_bar.haml

```diff
@@ -8,12 +8,17 @@
     function initialize() {
       var input = document.getElementById('q');
       var autocomplete = new google.maps.places.Autocomplete(input);
+      autocomplete.setTypes(['geocode']);
 
       google.maps.event.addListener(autocomplete, 'place_changed', function(event){
         var place = autocomplete.getPlace();
-        if(place != null && place.name != "") {
-          console.log('name:', place.name, '\naddress: ', place.formatted_address, '\ncoordinates: ', place.geometry.location.toUrlValue());
-          document.getElementById('lc').value = place.geometry.location.toUrlValue();
+        if(place != null) {
+          if( place.geometry != null ) {
+            console.log('name:', place.name, '\naddress: ', place.formatted_address, '\ncoordinates: ', place.geometry.location.toUrlValue());
+            document.getElementById('lc').value = place.geometry.location.toUrlValue();
+          } else {
+            console.log("Please select some place from the autocomplete list.")
+          }
         }
         document.getElementById('homepage-filters').submit();
```

Comment:
Yes! I should have left some comment that there are two things to do:
1) Add some error handling for situations where user has not selected any of the suggestions
2) Decide what data should be passed to Homepage controller

I was planning to do the first step at least before raising my hands off this part of the search. (i.e. new branch & more .js code there.)